server/src/inmantals/server.py

```diff
@@ -236,13 +379,18 @@ def treeify_reverse(iterator):
                 os.path.realpath(k): treeify_reverse(v) for k, v in groupby(anchormap, lambda x: x[1].file)
             }
 
+        async def sync_compile_and_anchor_folders() -> None:
+            for folder in self._workspace_folders.values():
+                # run synchronous part in executor to allow context switching while awaiting
+                await asyncio.get_event_loop().run_in_executor(self.threadpool, sync_compile_and_anchor_folder, folder)
+                await self.publish_diagnostics(None)
+                logger.info(f"Compilation succeeded for folder {folder}")
+
         try:
             if self.shutdown_requested:
                 return
+            await sync_compile_and_anchor_folders()
```

Comment:
So we always compile all folders? I would think there would always be one active one and we'd only need to compile that one? If you think this question is better suited for the other PR that's fine as well.

Reply:
Indeed, this is an artifact from the previous implementation. Only the "active" folder should get compiled.